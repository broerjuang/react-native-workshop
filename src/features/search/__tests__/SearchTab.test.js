import renderer from 'react-test-renderer';
import React from 'react';
import {
  SearchTab,
  mapStateToProps,
  mapDispatchToProps,
} from '../assets/SearchTab';

describe('SearchRepositoryScreen Test', () => {
  it('should render SearchRepositoryScreen correctly', () => {
    let component = renderer.create(
      <SearchTab
        navigation={{navigate: (s) => s}}
        handleClearSearch={() => {}}
        handleSearchRepo={() => {}}
        handleSearchUser={() => {}}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should mapStateToProps correctly', () => {
    let result = mapStateToProps({});
    expect(result).toEqual({state: {}});
  });

  it('should mapDispatchToProps correctly', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).handleSearchRepo('test');
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: 'SEARCH_REPOS',
      payload: {searchInput: 'test'},
    });

    mapDispatchToProps(dispatch).handleSearchUser('test');
    expect(dispatch.mock.calls[1][0]).toEqual({
      type: 'SEARCH_USERS',
      payload: {searchInput: 'test'},
    });

    mapDispatchToProps(dispatch).handleClearSearch();
    expect(dispatch.mock.calls[2][0]).toEqual({
      type: 'CLEAR_SEARCH',
    });
  });
});
