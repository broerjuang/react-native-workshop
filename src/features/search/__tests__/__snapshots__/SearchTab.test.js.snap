// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchRepositoryScreen Test should render SearchRepositoryScreen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#efefef",
        "borderRadius": 5,
        "display": "flex",
        "flexDirection": "row",
        "margin": 6,
        "paddingHorizontal": 6,
        "paddingVertical": 2,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[Function]}
      returnKeyType="search"
      style={
        Object {
          "backgroundColor": "#efefef",
          "height": 40,
          "width": "90%",
        }
      }
      value=""
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#f5f5f5",
          "borderColor": "#e3e3e3",
          "borderRadius": 3,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 40,
          "marginBottom": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 5,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "borderRightColor": "#bdc6cf",
            "borderRightWidth": 0,
          },
          false,
          false,
          Object {
            "borderBottomLeftRadius": 3,
            "borderTopLeftRadius": 3,
          },
          Object {
            "backgroundColor": "white",
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#5e6977",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                },
                undefined,
                Object {
                  "color": "#43484d",
                },
                undefined,
              ],
            ]
          }
        >
          Repositories
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          false,
          Object {
            "borderLeftColor": "#bdc6cf",
            "borderLeftWidth": 1,
          },
          Object {
            "borderBottomRightRadius": 3,
            "borderTopRightRadius": 3,
          },
          false,
          false,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#5e6977",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                },
                undefined,
                false,
                false,
              ],
            ]
          }
        >
          Users
        </Text>
      </View>
    </View>
  </View>
</View>
`;
